pyresample/geometry.py

```diff
@@ -982,26 +984,56 @@ def pixel_size_y(self):
             return None
         return self.resolution[1]
 
-    def compute_domain(self, corners, resolution=None, shape=None):
+    def compute_domain(
+            self,
+            corners: Sequence,
+            resolution: Optional[Union[float, tuple[float, float]]] = None,
+            shape: Optional[tuple[int, int]] = None,
+            projection: Optional[Union[CRS, dict, str]] = None
+    ):
         """Compute shape and area_extent from corners and [shape or resolution] info.
 
-        Corners represents the center of pixels, while area_extent represents the edge of pixels.
+        Args:
+            corners:
+                4-element sequence representing the outer corners of the
+                region. Note that corners represents the center of pixels,
+                while area_extent represents the edge of pixels. The four
+                values are (xmin_corner, ymin_corner, xmax_corner, ymax_corner).
+                If the x corners are ``None`` then the full extent (area of use)
+                of the projection will be used if defined. A RuntimeError will
+                be raised if the PROJ library does not know the area of use for
+                the projection and it is not a geographic lon/lat projection.
```

Comment:
This is slightly ambiguous.  It can mean that a RuntimeError will be raised if the PROJ library does not know the area of use, or it can mean that a RuntimeError will be raised if th PROJ library does not know the area of use AND the user passed ``None`` as the x corners.

Reply:
It is the AND case since the area of use is only used if the corners are `None`. I will try to clarify that.